modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/endpoint/liferay/LiferayOAuthDataProvider.java

```diff
@@ -1328,6 +1308,24 @@ private User _getUser(UserSubject userSubject) throws Exception {
 			GetterUtil.getLong(userSubject.getId()));
 	}
 
+	private void _handleInvalidRequest(
```

Comment:
Try to extract this method and reuse here and in the `LiferayDynamicRegistrationService`.